### PR TITLE
Exporter: Update elf thunk resolution

### PR DESCRIPTION
**Pull request by alessandrogario**
Fixes issue https://github.com/lifting-bits/anvill/issues/156

### CODE REVIEW

File: anvill/python/anvill/program.py [hunk 1]
```diff
@@ -175,6 +175,10 @@ def proto(self):
             proto["functions"].append(func.proto())
 
         for source, target in self._control_flow_redirections.items():
+            if type(source) != int or type(target) != int:
```
Comment (pgoodman) [inline anvill/python/anvill/program.py:178]: This is the wrong place to fix it; instead, we should not admit "bad data" into the object in the first place, i.e. `add_control_flow_redirection` should be annotated with types `from: int, to: int` and then it should be enforced at the source, and we should work our way back to the source of the issue.

**Review by pgoodman — CHANGES_REQUESTED** [review]

**Comment by kumarak** [comment]
The fix will bypass the `null` value but not resolve the actual problem. The null value creeps in because the relocation offset is getting wrong from the elf image.

**Comment by alessandrogario** [comment]
> The fix will bypass the `null` value but not resolve the actual problem. The null value creeps in because the relocation offset is getting wrong from the elf image.

There are two commits in the PR, the first one solves the address problem

**Review by kumarak — APPROVED** [review]

File: anvill/python/anvill/binja/bnprogram.py [hunk 2]
```diff
@@ -34,7 +34,11 @@ def __init__(self, bv: bn.BinaryView, path: str):
         self._path: Final[str] = path
         self._bv: Final[bn.BinaryView] = bv
         self._type_cache: Final[TypeCache] = TypeCache(self._bv)
-        self._init_ctrl_flow_redirections()
+
+        try:
+            self._init_ctrl_flow_redirections()
+        except:
+            print("Failed to initialize control flow redirections")
```
Comment (pgoodman) [inline anvill/python/anvill/binja/bnprogram.py:41]: File an issue that we eventually need a better logging approach.

**Review by pgoodman — APPROVED** [review]

File: anvill/python/anvill/binja/bnprogram.py (hunk 2 above)
Comment (alessandrogario) [reply anvill/python/anvill/binja/bnprogram.py]: Opened it here: https://github.com/lifting-bits/anvill/issues/166

**Review by pgoodman — APPROVED** [review]

**Pull request merged**